src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleConnectionParameters.java

```diff
@@ -114,6 +136,90 @@ private String generateUrl() {
       buf.append("/");
       buf.append(serviceName);
     }
+
     return buf.toString();
   }
+
+  private String generateSecureUrl(StringBuilder buf) {
+    return new OracleTNSEntryBuilder()
+        .withProtocol("TCPS")
+        .withHost(host)
+        .withPort(port)
+        .withInstanceName(instance)
+        .withServiceName(serviceName)
+        .build(buf);
+  }
+
+  private void checkInstanceAndServiceName() {
+    if (instance != null && serviceName != null) {
+      String errorMessage = "Instance (SID) : [" + instance + "] and Service Name : [" + serviceName
+          + "] were provided at the same time, please configure only one";
+
+      throw new ModuleException(errorMessage, CONNECTIVITY, new ConnectionException(errorMessage));
+    }
+  }
+
+  /**
+   *
+   * Builder class to facilitate the creation of Oracle's URLs that use TNS Entries
+   *
+   */
+  public static final class OracleTNSEntryBuilder {
```

Comment:
what's the purpose of this being public if the constructor is private and there's no method returning an instance of it. This seems better fit as a package scoped top level class which can be unit tested